packages/core/contracts/oracle/implementation/optimistic-arbitrator/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,181 @@
+            disputer: address(0),
+            currency: currency,
+            settled: false,
+            proposedPrice: 0,
+            reward: reward,
+            finalFee: _getStore().computeFinalFee(address(currency)).rawValue,
+            bond: bond,
+            customLiveness: customLiveness,
+            expirationTime: 0
+        });
+
+        if (reward > 0) currency.safeTransferFrom(msg.sender, address(this), reward);
+    }
+
+    function proposePrice(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        int256 proposedPrice
+    ) public {
+        Request storage request = requests[_getId(msg.sender, identifier, timestamp, ancillaryData)];
+        require(address(request.currency) != address(0), "Price not requested");
+        require(request.proposer == address(0), "Current proposal in liveness");
+
+        request.proposer = msg.sender;
+        request.proposedPrice = proposedPrice;
+        request.expirationTime = uint64(getCurrentTime()) + request.customLiveness;
+
+        request.currency.safeTransferFrom(msg.sender, address(this), request.bond + request.finalFee);
+    }
+
+    function disputePrice(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData
+    ) public {
+        Request storage request = requests[_getId(msg.sender, identifier, timestamp, ancillaryData)];
+        require(request.proposer != address(0), "No proposed price to dispute");
+        require(request.disputer == address(0), "Proposal already disputed");
+        require(uint64(getCurrentTime()) < request.expirationTime, "Proposal past liveness");
+
+        request.disputer = msg.sender;
+
+        request.currency.safeTransferFrom(msg.sender, address(this), request.bond + request.finalFee);
+
+        OptimisticOracleV2Interface oo = _getOptimisticOracle();
+        oo.requestPrice(identifier, timestamp, ancillaryData, request.currency, request.reward);
+        oo.proposePriceFor(
+            request.proposer,
+            address(this),
+            identifier,
+            timestamp,
+            ancillaryData,
+            request.proposedPrice
+        );
+        oo.disputePriceFor(msg.sender, address(this), identifier, timestamp, ancillaryData);
+
+        delete requests[_getId(msg.sender, identifier, timestamp, ancillaryData)];
+    }
+
+    function settleAndGetPrice(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData
+    ) public returns (int256) {
+        Request storage request = requests[_getId(msg.sender, identifier, timestamp, ancillaryData)];
+        require(address(request.currency) != address(0), "Price not requested");
+        require(request.proposer != address(0), "No proposed price to settle");
+        require(uint64(getCurrentTime()) > request.expirationTime, "Proposal not passed liveness");
+        require(request.disputer == address(0), "Proposal disputed, cant settle");
```

Comment:
If a price request is contested and the disputant loses. The proposer would have to call the `OOv2.settleAndGetPrice()` contract to get the bond back, right? 
Would it make sense to add a function in this contract to help with this task?